Rubberduck.Parsing/VBA/RubberduckParserState.cs

```diff
@@ -977,6 +984,31 @@ public void AddAttributesRewriter(QualifiedModuleName module, IModuleRewriter at
             _moduleStates[key].SetAttributesRewriter(attributesRewriter);
         }
 
+        private List<object> _queuedRequestors;
+        public void SuspendParser(object requestor, Action busyAction)
+        {
+            if (Status != ParserState.Ready)
+            {
+                throw new InvalidOperationException("Cannot suspend the parser while it is running. Either cancel or wait for Ready status");
```

Comment:
This is not safe for several reasons, the basic problem being that this is not thread-safe.

Reply:
To clarify - are we saying that `throw` shouldn't be performed here? What would be the recommendation here?